app/services/courses/assign_program_type_service.rb

```diff
@@ -0,0 +1,24 @@
+module Courses
+  class AssignProgramTypeService
+    def execute(funding_type, course)
+      case funding_type
+      when "salary"
+        if !course.self_accredited?
```

Comment:
```suggestion
        unless course.self_accredited?
```

Reply:
IMO inverting the if is better than doing an `unless`, I find unless is really jarring to try and parse because it feels like hidden negation